web/src/components/tasks/search-business-name/SearchBusinessNameForm.tsx

```diff
@@ -174,42 +234,55 @@ export const SearchBusinessNameForm = (props: Props): ReactElement => {
         </WithErrorBar>
       )}
       <div className="margin-top-2">
-        {nameAvailability?.status === "AVAILABLE" && (
-          <Available submittedName={submittedName} updateButtonClicked={updateButtonClicked} />
+        {props.nameAvailability?.status === "AVAILABLE" && (
+          <Available submittedName={props.businessName} updateButtonClicked={updateButtonClicked} />
         )}
-        {nameAvailability?.status === "DESIGNATOR_ERROR" && (
+        {props.nameAvailability?.status === "DESIGNATOR_ERROR" && (
           <Alert variant="error" dataTestid="designator-error-text">
             <p className="font-sans-xs">{Config.searchBusinessNameTask.designatorText}</p>
           </Alert>
         )}
-        {nameAvailability?.status === "SPECIAL_CHARACTER_ERROR" && (
+        {props.nameAvailability?.status === "SPECIAL_CHARACTER_ERROR" && (
           <Alert variant="error" dataTestid="special-character-error-text">
             <Content className="font-sans-xs">
               {templateEval(Config.formation.fields.businessName.alertSpecialCharacters, {
-                name: submittedName,
+                name: props.businessName,
               })}
             </Content>
           </Alert>
         )}
-        {nameAvailability?.status === "RESTRICTED_ERROR" && (
+        {props.nameAvailability?.status === "RESTRICTED_ERROR" && (
           <Alert variant="error" dataTestid="restricted-word-error-text">
             <Content className="font-sans-xs">
               {templateEval(Config.formation.fields.businessName.alertRestrictedWord, {
-                name: submittedName,
-                word: nameAvailability.invalidWord ?? "*unknown*",
+                name: currentName,
+                word: props.nameAvailability.invalidWord ?? "*unknown*",
               })}
             </Content>
           </Alert>
         )}
 
-        {nameAvailability?.status === "UNAVAILABLE" && (
+        {props.nameAvailability?.status === "UNAVAILABLE" && (
           <Unavailable
             resetSearch={(): void => {
               resetSearch();
-              setNameAvailability(undefined);
+              if (updateQueue) {
+                updateQueue
+                  .queueFormationData({
+                    formationFormData: {
+                      ...updateQueue.current().formationData.formationFormData,
+                    },
+                    businessNameAvailability: emptyNameAvailability,
+                    dbaBusinessNameAvailability: emptyNameAvailability,
+                  })
+                  .queueProfileData({
+                    nexusDbaName: "",
+                  })
+                  .update();
+              }
```

Comment:
A function this large shouldn't be inline in JSX and instead should be extracted as its own named function

Reply:
I agree. I wasn't super concerned about this because I thought I was going to immediately handle it with the refactor of BusinessNameSearch that I was trying to tackle following this. Not really an excuse for keeping it like this though.